### PR TITLE
Improves the docs of config/runtime.exs and Config.config_env/0

### LINKED ISSUE #10413: Unclear docs about config/runtime.exs
The current docs for `config/runtime.exs` are

> For runtime configuration, you can use the config/runtime.exs file. It is executed after your Mix project is compiled and also before a release (assembled with mix release) starts.

I do not understand this sentence:

> It is executed after your Mix project is compiled

when a project is compiled, as in `mix compile` or when building a Docker image, anything executed is executed in the void, you see what I mean :).

Does that mean "before the application starts"?

### PR DESCRIPTION
**Pull request by fxn**
Fixes #10413.

### CODE REVIEW

File: lib/elixir/lib/config.ex [hunk 1]
```diff
@@ -160,7 +160,8 @@ defmodule Config do
   end
 
   @doc """
-  Returns the environemnt this configuration file is executed on.
+  In Mix projects this function returns the environment this configuration
```
Comment (josevalim) [inline lib/elixir/lib/config.ex:163]: The first line should always be a short summary, as that is shown in tables and stuff, so what do you think about this?

```suggestion
  Returns the environment this configuration file is executed on.

  In Mix projects this function returns the environment this configuration
```

Comment (fxn) [reply lib/elixir/lib/config.ex:163]: Ah, forgot that! Sure, let me rebase in a while!

Comment (josevalim) [reply lib/elixir/lib/config.ex:163]: There is no need to rebase, we always squash it anyway. :) Thanks!

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Comment by fxn** [comment]
❤️

**Pull request merged**